e2e/testcafe-devextreme/tests/dataGrid/common/markup/T1163515_alternateRowGroupBorders.ts

```diff
@@ -0,0 +1,291 @@
+    const rowsViewBorderRightColor = await rowsView.getStyleProperty('border-right-color');
+    const rowsViewBorderBottomColor = await rowsView.getStyleProperty('border-bottom-color');
+
+    await t.expect(FORBIDDEN_COLORS).notContains(rowsViewBorderTopColor.toLowerCase());
+    await t.expect(FORBIDDEN_COLORS).notContains(rowsViewBorderLeftColor.toLowerCase());
+    await t.expect(FORBIDDEN_COLORS).notContains(rowsViewBorderRightColor.toLowerCase());
+    await t.expect(FORBIDDEN_COLORS).notContains(rowsViewBorderBottomColor.toLowerCase());
+  }
+
+  if (matrixOptions.rowAlternationEnabled) {
+    const filteredRows = dataGrid.getRows().find(`tr:not(.${SELECTORS.masterDetailClass})`);
+    const filteredRowsLength = await filteredRows.count;
+
+    let i = 1;
+    while (i < filteredRowsLength) {
+      const currentRow = filteredRows.nth(i);
+      const previousRow = filteredRows.nth(i - 1);
+
+      const currentClasses = await currentRow.getAttribute('class');
+      const previousClasses = await previousRow.getAttribute('class');
+
+      if (currentClasses?.includes(SELECTORS.groupRowClass)) {
+        i += 2;
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+
+      if (previousClasses?.includes(SELECTORS.groupRowClass)) {
+        i += 1;
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+
+      if (currentClasses !== previousClasses) {
+        const currentBg = await currentRow.getStyleProperty('background-color');
+        const previousBg = await previousRow.getStyleProperty('background-color');
+
+        await t.expect(currentBg).notEql(previousBg);
+      }
+
+      i += 1;
+    }
+  }
+
+  if (matrixOptions.showRowLines) {
+    const { dataRows } = dataGrid;
+    const dataRowsCount = await dataRows.count;
+
+    for (let i = 0; i < dataRowsCount; i += 1) {
+      const dataRow = dataRows.nth(i);
+
+      const borderTop = await dataRow.getStyleProperty('border-top-width');
+      const borderTopStyle = await dataRow.getStyleProperty('border-top-style');
+
+      await t.expect(parseInt(borderTop, 10)).gt(0);
+      await t.expect(borderTopStyle).notEql('none');
+
+      if (!matrixOptions.showBorders) {
+        const borderBottom = await dataRow.getStyleProperty('border-bottom-width');
+        const borderBottomStyle = await dataRow.getStyleProperty('border-bottom-style');
+
+        await t.expect(parseInt(borderBottom, 10)).gt(0);
+        await t.expect(borderBottomStyle).notEql('none');
+      }
+    }
+  }
+
+  if (matrixOptions.showColumnLines) {
+    const cells = dataGrid.getCells().filter('td:not([class])');
+    const cellsCount = await cells.count;
+
+    for (let i = 0; i < cellsCount; i += 1) {
+      const cell = cells.nth(i);
+
+      const borderRightWidth = await cell.getStyleProperty('border-right-width');
+      const borderRightColor = await cell.getStyleProperty('border-right-color');
+
+      await t.expect(borderRightWidth).eql('1px');
```

Comment:
You also need to consider the cases for the first/last and fixed cells. Their styles will differ from those of the other columns.

Reply:
Added check border size (2 or 1px) for cases with fixed columns